### DISCUSSION TITLE
Clear-PnPRecycleBinItem  List View Threshold when using the -Identity parameter

### DISCUSSION THREAD

**Discussion by brodemann** [pnp/powershell#1063] [Q&A]
### Reporting an Issue or Missing Feature
Issue

### Expected behavior 
Deletion of item from Recycle Bin using the -Identity parameter

### Actual behavior 
Getting list view threshold error

Error:
Clear-PnpRecycleBinItem : The attempted operation is prohibited because it exceeds the list view threshold.
At line:1 char:1
+ Clear-PnpRecycleBinItem -Identity $ItemId -Force -ErrorVariable Delet ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Clear-PnPRecycleBinItem], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.RecycleBin.ClearRecycleBinItem

### Steps to reproduce behavior
ForEach($item in $bin){
     $itemId = $item.Id.Guid
     Clear-PnpRecycleBinItem -Identity $itemId -Force
}

### What is the version of the Cmdlet module you are running?
1.6.0

### Which operating system/environment are you running PnP PowerShell on?
Windows

**Comment by latchee** [comment]
Hi @brodemann,

I know this post is old but I am having the same error as you currently. Have you ever found a work around?

Kind regards,

**Reply by Barbarur** [reply]
Using REST API seems to be the way to go:


https://pnp.github.io/script-samples/spo-bulk-delete-recyclebin-in-batch-avoid-lvt/README.html?tabs=pnpps

**Reply by latchee** [reply]
Thank you for you quick response! 

If that can help anybody in the future. I replaced:
`# Retrieve items from the recycle bin
$recycleBinItems = Get-PnPRecycleBinItem | Where-Object { $_.DeletedByName -eq $User }

# Delete items
    foreach ($item in $recycleBinItems) {
        Clear-PnPRecycleBinItem -Identity $item -Force
        Write-Host "$($item.Title) deleted."
    }
`

With:
`# Retrieve items from the recycle bin
$recycleBinItems = Get-PnPRecycleBinItem | Where-Object { $_.DeletedByName -eq $User }

# Delete items
    foreach ($item in $recycleBinItems) {
        $requestUrl = $SiteURL + "/_api/site/RecycleBin/DeleteByIds"
        $requestBody = "{""ids"":[""$($item.id)""]}"

        Invoke-PnPSPRestMethod -Method POST -Url $requestUrl -Content $requestBody | Out-Null
        Write-Host "$($item.Title) deleted."
`

Edits: rendering of the code is weird.
